src/lib/build/mod.rs

```diff
@@ -1,5 +1,69 @@
 //! Actually building Idris packages.
+//!
+//! The build process contains two major phases: dependency preparation and target generating.
```

Comment:
Wow, I've not even heard about the task system of Cargo before. I've read the post just now, then I think it will be just a system that is too big to be implemented in our 0.1.0 release. 

The idea of tasks is basically a more flexible custom subcommand intergration, and I think we are not yet ready for it now.

> I don't think we need to copy the dependencies into target/deps.

The deps folder is mainly for ide plugins to import dependencies at edit-time, or else we have to cook a new meta file which refers the path to global build cache and we have to teach each editor plugin to read it. It's a hard job for us, I think.

Reply:
Tasks would be a fairly simple addition to the target generation step I think (just make binaries for declared tasks iff the current root package is local and use the `target` dir to build/store them), but I agree that it isn't really necessary for a 0.1.0 release.

If an IDE wants to see our dependencies (and either they don't want to use the Idris IDE server or the server doesn't let us add import paths), maybe we could symlink into `target/deps` instead of copying the files? Constantly copying deps back and forth doesn't seem super efficient or kind for disk space...

Reply:
My solution is also future compatible with tasks, since the task subcommand can simply compile the task executable under the root pacakge folder, for example `targrt/tasks/`.

Symlink is also an available option, but AFAIN, the symlink system in Window is shitty. However, we should only setup the deps/ folder for root package so it seems not to be a big problem on performance and disk space.